### PR TITLE
feat!: Simplify tool typing and add content/text generation conversion utilities

### PR DESCRIPTION
**Pull request by toyamarinyon**
## Overview

This PR simplifies the language model tool typing system and introduces a conversion layer to support migration between legacy `TextGenerationNode` and new `ContentGenerationNode` formats. The main motivation is to streamline the tool configuration approach by removing redundant schema definitions and providing robust conversion utilities for backward compatibility during the transition period.

### Breaking Changes ⚠️
- **Removed `configurationSchema`** from `LanguageModelTool` interfaces - configurations now rely on per-option schemas instead
- **Dropped `Schema` generic type parameter** across all tool-related types (`LanguageModelToolWithoutTools`, `LanguageModelToolWithTools`, `LanguageModelTool`, `defineLanguageModelTool`, `hasTools`)
- **Removed `hasConfigurationSchema` function** export

### Core Improvements
- **Simplified tool typing** in `language-model-registry`:
  - Streamlined generics and overloads for `defineLanguageModelTool` and `hasTools`
  - Re-exported tool types from `tools.ts` for easier imports
  
- **Added conversion utilities** in `node-registry`:
  - `convertTextGenerationToContentGeneration` - migrates legacy format to new format
  - `convertContentGenerationToTextGeneration` - provides backward compatibility
  - Intelligent language model ID mapping across providers (OpenAI, Google, Anthropic) with fallback strategies:
    - claude-opus-4.5 → 4.1
    - gemini-3-pro-preview → 2.5-pro
    - gpt-5.1-* → gpt-5 variants
    - sonar → openai/gpt-5-nano
  - Tool configuration conversion between schemas (github-api, postgres, openai-web-search, anthropic-web-search)
  - Secret validation and mapping

- **Enhanced type exports** from `language-model`:
  - Exported `AnthropicLanguageModelId`, `GoogleLanguageModelId`, and `OpenAILanguageModelId` for parsing/validation support

## Testing

- Added comprehensive Vitest test suite to `node-registry` with:
  - Round-trip conversion tests verifying data integrity
  - Model ID mapping validation across all providers
  - Tool configuration conversion tests
  - Secret handling verification
  - Test fixtures for realistic conversion scenarios
- All existing tests pass without modification

## Review Notes

Please pay special attention to:
- **Breaking changes impact** - Verify that removing `configurationSchema` aligns with our deprecation timeline
- **Model ID mapping logic** - Confirm the fallback strategies for newer model versions are appropriate
- **Conversion completeness** - Review if any edge cases in the conversion utilities need additional handling

## Related Issues

_No specific issues referenced - this is part of the ongoing migration from TextGenerationNode to ContentGenerationNode format_

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Simplified tool typing by removing `Schema` generic and `configurationSchema` property

- Added bidirectional conversion utilities between `TextGenerationNode` and `ContentGenerationNode` formats

- Implemented intelligent language model ID mapping with fallback strategies across providers

- Added comprehensive test suite with round-trip conversion validation and tool configuration tests

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["TextGenerationNode<br/>Legacy Format"] -->|convertTextGenerationToContentGeneration| B["ContentGenerationNode<br/>New Format"]
  B -->|convertContentGenerationToTextGeneration| A
  C["Tool Configs<br/>github, postgres, openai-web-search"] -->|Convert| D["New Tool Format<br/>with standardized schema"]
  E["Language Model IDs<br/>with version mapping"] -->|Map| F["Provider/ID pairs<br/>with fallbacks"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Breaking change</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>tool.ts</strong><dd><code>Remove Schema generic and configurationSchema property</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-3af907fbd24f45919b0812358e56ae383a40f4727ad894647c2c9e0b3b1ec070">+9/-40</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>tools.ts</strong><dd><code>Re-export tool types for easier imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-cf3671d339b69de221b03d748bed0a3fc7a843323730a7e99d10662544a2e3ac">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Export language model ID types for validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-4ccf7fe729d450f766d2a75fdd01f246d9f6a16f9386d58a4afca8b2f2c2ce23">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>node-convertion.ts</strong><dd><code>Add bidirectional node format conversion utilities</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-306e8341813bac6c832888cbbbff44cad1945c2867da2edfbe116c13f5099fa6">+315/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>nodes.ts</strong><dd><code>Add test fixtures for node conversion scenarios</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-73361fde4253eff88457bdbe77fa836aa994a7fe32a1658700aaa490645d83c8">+197/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>node-convertion.test.ts</strong><dd><code>Add comprehensive conversion and round-trip tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-099b46cd07a69bce0233f5ca2e1a09eb7e469127d56fe732de651137e2423068">+298/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>package.json</strong><dd><code>Add vitest dependency and test script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-d4aa3db3ba8de780210335fe3254d46f5ff9c0ae3cff0a930fe3e41d434bfd66">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>packages-license.md</strong><dd><code>Update license documentation with dependency changes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-ae1da9a37e22d3531ffedff27e1dd92e0c755e79f6062620c4370f8dde00b2ab">+2/-46</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>pnpm-lock.yaml</strong><dd><code>Update lock file with vitest and platform-specific dependencies</code></dd></td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/2280/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+36/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

> [!NOTE]
> Simplifies language-model tool types (drops schema generics) and adds utilities/tests to convert between TextGenerationNode and ContentGenerationNode with model/tool mapping.
> 
> - **language-model-registry**
>   - **Tool typing simplified**: Remove `Schema` generic and `configurationSchema` from `LanguageModelTool*` types; delete `hasConfigurationSchema`; update overloads for `defineLanguageModelTool` and `hasTools`.
>   - **Exports**: Re-export tool types from `tools.ts`.
> - **language-model**
>   - **Type exports**: Export `AnthropicLanguageModelId`, `GoogleLanguageModelId`, `OpenAILanguageModelId`.
> - **node-registry**
>   - **Conversion utilities**: Add `convertTextGenerationToContentGeneration` and `convertContentGenerationToTextGeneration` with provider-aware model ID mapping and fallbacks.
>   - **Tool mapping**: Convert `github-api`, `postgres`, `openai-web-search`, `anthropic-web-search`; validate `SecretId`.
>   - **Tests**: Add Vitest suite (fixtures and round-trip tests); add `test` script in `package.json`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 501715a383a86ee61155f30555182fe53bb75e7e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added bidirectional conversion functionality between text generation node formats, supporting OpenAI, Anthropic, and Google models with comprehensive tool integration capabilities.

* **Tests**
  * Introduced extensive test suite covering node conversion across multiple language model providers with various tool configurations and edge cases.

* **Chores**
  * Updated package license documentation with Mozilla Public License 2.0 entry and dependency cleanup.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/node-registry/src/node-convertion.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+				if (result.error) {
+					continue;
+				}
+
+				const useTools = tool.configuration.useTools;
+				if (!Array.isArray(useTools)) {
+					continue;
+				}
+				tools.postgres = {
+					secretId: tool.configuration.secretId,
```
[the remark this answers is not in the document]
Comment (shige) [reply packages/node-registry/src/node-convertion.ts:276]: ref: https://github.com/giselles-ai/giselle/issues/2282

File: packages/node-registry/src/node-convertion.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+					tools: (tool.configuration.useTools as string[]) ?? [],
+				};
+				break;
```
[the remark this answers is not in the document]
Comment (shige) [reply packages/node-registry/src/node-convertion.ts:279]: ref: https://github.com/giselles-ai/giselle/issues/2282

**Review by shige — APPROVED** [review]
Thank you! 🧰

**Comment by toyamarinyon** [comment]
i'll fix some reviewes from review agents before merge

**Comment by toyamarinyon** [comment]
The points raised in the review have been marked as issues, so they will be merged.
